### PR TITLE
Use uid_t to deal with uids instead of integers

### PR DESCRIPTION
**Pull request by AtariDreams**
We can also use strtoul to convert strings since uid_t is unsigned.

### CODE REVIEW

File: backend/ipp.c [hunk 1]
```diff
@@ -1914,7 +1911,7 @@ main(int  argc,				/* I - Number of command-line args */
 
     if (ipp_status <= IPP_STATUS_OK_CONFLICTING && argc > 6)
     {
-      fprintf(stderr, "PAGE: 1 %d\n", copies_sup ? atoi(argv[4]) : 1);
+      fprintf(stderr, "PAGE: 1 %ld\n", copies_sup ? strtol(argv[4], NULL, 10) : 1);
```
Comment (michaelrsweet) [inline backend/ipp.c:1914]: As "copies" is limited to a 32-bit integer (IPP syntax), and more specifically to a range of 1 to 999 with the default configuration, I'd rather stick with atoi.

File: backend/ipp.c [hunk 2]
```diff
@@ -567,12 +565,13 @@ main(int  argc,				/* I - Number of command-line args */
 #endif /* HAVE_LIBZ */
       else if (!_cups_strcasecmp(name, "contimeout"))
       {
+        int value_int = (int)strtol(value, NULL, 10);
```
Comment (michaelrsweet) [inline backend/ipp.c:568]: Avoid casting strtol to int.

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
This PR contains a lot more changes than using uid_t.

**Comment by AtariDreams** [comment]
All done @michaelrsweet!

File: scheduler/cups-exec.c [hunk 3]
```diff
@@ -154,7 +154,7 @@ main(int  argc,				/* I - Number of command-line args */
   {
     cups_file_t	*fp;			/* File */
     char	line[1024];		/* Line from file */
-    int		linenum = 0;		/* Line number in file */
+    unsigned	linenum = 0;		/* Line number in file */
```
Comment (michaelrsweet) [inline scheduler/cups-exec.c:157]: OK, didn't notice this change before, but I'm not sure why you are making it?

File: scheduler/cups-exec.c [hunk 4]
```diff
@@ -164,8 +164,7 @@ main(int  argc,				/* I - Number of command-line args */
     {
       while (cupsFileGets(fp, line, sizeof(line)))
       {
-        linenum ++;
-        fprintf(stderr, "DEBUG: %4d  %s\n", linenum, line);
+        fprintf(stderr, "DEBUG: %4u  %s\n", ++linenum, line);
```
Comment (michaelrsweet) [inline scheduler/cups-exec.c:167]: I prefer leaving the increment outside of the printf.

File: scheduler/ipp.c [hunk 5]
```diff
@@ -7874,13 +7874,14 @@ get_subscriptions(cupsd_client_t  *con,	/* I - Client connection */
   }
   else if (!strncmp(resource, "/jobs/", 6) && resource[6])
   {
+    int resourceId = atoi(resource + 6);
```
Comment (michaelrsweet) [inline scheduler/ipp.c:7877]: This should be `job_id` since we are getting the job ID...

File: scheduler/main.c [hunk 6]
```diff
@@ -1435,8 +1435,7 @@ process_children(void)
     * Delete certificates for CGI processes...
     */
 
-    if (pid)
-      cupsdDeleteCert(pid);
+    cupsdDeleteCert(pid);
```
Comment (michaelrsweet) [inline scheduler/main.c:1438]: I think we want to keep the PID check - PID 0 is the root certificate.

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
Some more nits.

File: scheduler/cups-exec.c (hunk 3 above)
Comment (AtariDreams) [reply scheduler/cups-exec.c:157]: Linenum is never negative.

File: scheduler/main.c (hunk 6 above)
Comment (AtariDreams) [reply scheduler/main.c:1438]: The while loop mandates that pid > 0.

Unless this was a bug in which case we should be changing the while condition to accept >= 0

Comment (AtariDreams) [reply scheduler/main.c:1438]: Perhaps you made a mistake in this while loop, in either case?

Comment (AtariDreams) [reply scheduler/main.c:1438]: If so, I would like to correct this

**Comment by AtariDreams** [comment]
@michaelrsweet See my comments and I think this is good now

**Review by michaelrsweet — APPROVED** [review]

**Pull request merged**